packages/skeleton-svelte/src/lib/components/Listbox/ListboxItem.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+	import { createId } from '$lib/internal/create-id.js';
+	import { getListboxContext } from './context.js';
+	import { getFirstElement, getLastElement, getNextElement, getPreviousElement } from '$lib/internal/elements.js';
+	import type { ListboxItemProps } from './types.js';
+
+	// Props
+	let {
+		value,
+		base = 'flex items-center gap-2 text-left',
+		padding = 'py-2 px-4',
+		radii = 'rounded',
+		selected = 'preset-filled',
+		classes,
+		leadBase,
+		leadClasses,
+		trailBase,
+		trailClasses,
+		children,
+		lead,
+		trail,
+		...attributes
+	}: ListboxItemProps = $props();
+
+	// Context
+	const context = getListboxContext();
+
+	// Local
+	const id = createId();
+	const listboxItemSelector = `[data-skeleton-id="${context.id}"] > [data-skeleton-part="listbox-item"]`;
+
+	// Events
+	const onclick: (typeof attributes)['onclick'] = (event) => {
```

Comment:
I've noticed it can feel a bit weird mixing passed event props (ex: `onclick`) and implementing local functions (ex: also called `onclick` here). It's not immediately clear the source of these, and could lead to some potential naming conflicts.

For other components I've opted for the following conventions...

Passed event props adhere to the framework preference:

- `onclick` for Svelte; `onClick` for React
- `onkeydown` for Svelte; `onKeyDown` for React

Then for the local functions we can set our own standard since this is all internal:

- `function onClickHandler() { ... }`
- `function onKeyDownHandler() { ... }`

> Camel case + "handler" at the end.

This is just one idea for the convention, but I'm open to other suggestions.

Reply:
I'd also welcome some insight as why everyone is moving from:

`function doSomething() {}`

To:

`const doSomething = () => {}`

Wasn't sure if it was just personal preference, but last I looked into things scope works different between standard and arrow functions.

Reply:
First point:
The reason I named it `onclick` and not `onclickHandler` or whatever is so that when applying it to the component I can use Svelte's shorthand: `{onclick}` instead of doing: `onclick={onClickHandler}` which is much more verbose. I can change it though, that's not a problem at all, just explaining my reasoning, lmk how you feel about this.

Reply:
Second point:
Using `const` for arrow functions makes code more maintainable in my experience for 3 reason:
1. Arrow functions don't hoist, which is confusing behaviour
2. Arrow functions don't inherit `this` which is also a source of confusion
3. Arrow functions are more concise and less verbose when overloading

So mainly a preference, but for good reasons IMHO.

Reply:
Yeah, to put it another way. Imagine I had a complete template that implemented:

```html
<!-- SomeComponent.svelte -->
<button {onclick}>...</button>
```

Without seeing more of the component you don't know if the source is internal or external. Even if it's more verbose, being able to quickly identify that would be handy. We can likely still opt for the shorthand on prop provided event handlers.

Plus if you need to trigger the prop provided event handler inside a local function, you might end up with:

```
function onclick() {
    // ...
    onclick?.();
}
```
Which would fail because the names conflict.

Reply:
I'm actually already doing this through `...attributes`:
```js
const onclick: (typeof attributes)['onclick'] = (event) => {
    ctx.isSelected(value) ? ctx.deselect(value) : ctx.select(value);
    attributes.onclick?.(event); // External 
};

Reply:
First I've seen this. Is `attributes` similar to `$$props` from Svelte 4?

Reply:
This is similar to $$restProps